scapy/sendrecv.py

```diff
@@ -64,143 +105,113 @@ def sndrcv(pks, pkt, timeout = None, inter = 0, verbose=None, chainCC=0, retry=0
     hsent={}
     for i in tobesent:
         h = i.hashret()
-        if h in hsent:
-            hsent[h].append(i)
-        else:
-            hsent[h] = [i]
+        hsent.setdefault(i.hashret(), []).append(i)
+
     if retry < 0:
         retry = -retry
-        autostop=retry
+        autostop = retry
     else:
-        autostop=0
-
+        autostop = 0
 
     while retry >= 0:
-        found=0
-    
+        found = 0
+
         if timeout < 0:
             timeout = None
-            
-        rdpipe,wrpipe = os.pipe()
-        rdpipe=os.fdopen(rdpipe)
-        wrpipe=os.fdopen(wrpipe,"w")
 
-        pid=1
+        rdpipe, wrpipe = os.pipe()
+        rdpipe = os.fdopen(rdpipe)
+        wrpipe = os.fdopen(wrpipe,"w")
```

Comment:
For multiprocessing, you may use
`multiprocessing.Pipe(False)` (uni-directional) to get `Connection` objects.
You can select those or, on windows, use the `poll` function to replace the current pipes